kpi-tracker/src/main.rs

```diff
@@ -793,35 +820,77 @@ async fn process_payday_block(
     if !is_payday_block {
         return Ok(None);
     }
-
-    let tokenomics_info = node
-        .get_tokenomics_info(block_hash)
-        .await
-        .with_context(|| format!("Could not get tokenomics info for block: {block_hash}"))?
-        .response;
-
-    let RewardsOverview::V1 {
-        total_staked_capital,
-        ..
-    } = tokenomics_info else {
-        return Ok(None);
-    };
-
     let mut baker_response = node.get_bakers_reward_period(block_hash).await?.response;
 
     let mut baker_count = 0;
+    let mut open_baker_count = 0;
+    let mut closed_baker_count = 0;
+    let mut open_delegation_recipient_count = 0;
+    let mut closed_for_new_delegation_recipient_count = 0;
     let mut finalizer_count = 0;
+    let mut total_actively_delegated = 0;
+    let mut total_equity_capital = 0;
+    let mut delegating_accounts = HashSet::new();
     while let Some(baker) = baker_response.try_next().await? {
         baker_count += 1;
         if baker.is_finalizer {
             finalizer_count += 1;
         }
+        total_actively_delegated += baker.delegated_capital.micro_ccd() as i64;
+        total_equity_capital += baker.equity_capital.micro_ccd() as i64;
+
+        let pool_info = node
+            .get_pool_info(block_hash, baker.baker.baker_id)
+            .await?
+            .response
+            .pool_info;
+        match pool_info.open_status {
+            OpenStatus::OpenForAll => {
+                open_baker_count += 1;
+                if baker.delegated_capital.micro_ccd() != 0 {
+                    open_delegation_recipient_count += 1;
+                }
+            }
+            OpenStatus::ClosedForNew => {
+                if baker.delegated_capital.micro_ccd() != 0 {
+                    closed_for_new_delegation_recipient_count += 1;
+                }
+            }
+            OpenStatus::ClosedForAll => closed_baker_count += 1,
+        }
+
+        let mut delegators = node
+            .get_pool_delegators_reward_period(block_hash, baker.baker.baker_id)
+            .await?
+            .response;
+        while let Some(delegator) = delegators.try_next().await? {
+            delegating_accounts.insert(delegator.account);
```

Comment:
An account can only delegate to a single pool, so we don't need this set to keep track of them. We can just count them.